And/teams features

andrey-canon opened the pull request:
## Description 
Add the functionality to work with teams 
- A member's team can see the files that other members have uploaded .
- A Staff member get the original view where has the option to see all the uploads 
- If a member's team shares the link's file with another member of a different team, the last one will not be able to see that

## Reviewers 
- [ ] @jfavellar90 
- [ ] @diegomillan

jfavellar90 commented on edx_sga/api_teams.py:
```diff
@@ -0,0 +1,70 @@
+from requests_oauthlib import OAuth2Session
+
+LOG = logging.getLogger(__name__)
+os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
```
Why you had to set this env var?

andrey-canon replied:
this allows to use http
 https://stackoverflow.com/questions/27785375/testing-flask-oauthlib-locally-without-https

diegomillan commented on edx_sga/static/js/src/sga_team_view.js:
```diff
@@ -0,0 +1,99 @@
+    var dataState;
+    var block;
+
+    function renderTeamGrading(state){
```
are these functions copied from the sga original view???

andrey-canon replied:
yes, they are copied but have little modifications

diegomillan reviewed (approved):
Looks good to me given we already tested rocketchat api module.

jfavellar90 commented on edx_sga/sga.py:
```diff
@@ -1015,6 +1072,82 @@ def runtime_user_is_staff(self):
+        """
+        members = self.get_teams_members()
+        if not members:
+            return {"assignments":{}}
```
Should this return an empty list instead of returning an empty dict?

andrey-canon replied:
you are right

andrey-canon committed:
Addressing PR comments
Add css and implementing user image
Filter assignments view to the members team
Add new methods to handle the team view
Add new template
Add new javascript
Add api for teams

jfavellar90 commented on edx_sga/sga.py:
```diff
@@ -1079,7 +1079,7 @@ def team_grading_data(self):
         members = self.get_teams_members()
         if not members:
-            return {"assignments":{}}
+            return {"assignments":list()}
```
or [ ] ?
